File: javascript/math-functions/decimal_binary.js
function decToBin(num){
	var binary = [];

	var half = parseInt(num / 2);
	var i = 0;

	while(Math.pow(2, i) <= half){ i++; }

	while(i >= 0){
		if(num - (Math.pow(2,i)) >= 0){
			num -= Math.pow(2,i);
			binary.push(true);
		}
		else{
			binary.push(false);
		}
		i--;
	}
	return binary;
}
function binToDecimal(binary){
	var decimal = 0;
	for(var i = 0, e = binary.length - 1; i <= binary.length; i++, e--){
		if(binary[i] == true){
			decimal += parseInt(Math.pow(2, e));
		}
	}
	return decimal;
}
function printBin(binary){
	var bin = "";
	for(var i = 0; i <= binary.length; i++){
		if(binary[i] == true){
			bin += "1";
		}
		else{
			bin += "0";
		}
	}
	console.log(bin);
}

var x = 142;
var bin = decToBin(x);
var reverse = binToDecimal(bin);
console.log(x);
console.log(bin);
console.log(reverse);
printBin(bin);

File: java/math-functions/DecimalBinary.java
package powermod;

public class DecimalBinary {
	public static boolean[] decToBin(int num) {
		int half = (int) num / 2, i = 0;
		
		while((int)Math.pow(2, i) <= half)
			i++;
		
		boolean[] bin = new boolean[i+1];
		
		for(int j = 0; i >= 0; i--, j++) {
			if(num - (int)Math.pow(2, i) >= 0) {
				num -= (int)Math.pow(2, i);
				bin[j] = true;
			}
			else
				bin[j] = false;
		}
		
		return bin;
	}
	public static int binToDec(boolean[] bits) {
		int decimal = 0;
		int e = bits.length - 1;
		for(boolean bit: bits) {
			if(bit)
				decimal += (int) Math.pow(2, e);
			e--;
		}
		return decimal;
	}
	public static void printBin(boolean[] bits) {
		for(boolean bit: bits)
			if(bit) 
				System.out.print("1");
			else 
				System.out.print("0");
		System.out.println();
	}
}
